### PR TITLE
Fix dismiss button; See: websharks/comment-mail#193

### PR DESCRIPTION
**Pull request by kristineds**
Fix dismiss button; 

See: websharks/comment-mail#193

### CODE REVIEW

**Comment by kristineds** [comment]
@raamdev @jaswsinc Submitting my PR for this but it still needs some improvement. I would need your expertise on this. :) https://github.com/websharks/comment-mail-pro/commit/fb9dc1237de6376a62330a1e5e0e2da7b79f8c8d

So far, I got this (see below). 
screen shot 2016-02-25 at 10 06 40 pm

**Comment by jaswsinc** [comment]
I took a closer look at this while I was working on the WP Sharks Core recently. It seems the `is-dismissible` class is enough to get this working; i.e., WordPress core contains JavaScript that adds the dismiss button whenever that class is present. So adding the button markup is unnecessary, and it will actually result in two buttons if we do that.

Having said that, there is another problem...

The `is-dismissible` class and button that WP core creates only works as a way to hide the current notice from the DOM. It offers no onclick handler or any back-end interaction that would allow us to actually clear a persistent notice from the DB.

For that reason, I had to take a slightly different approach in the WP Sharks Core. Instead of adding the `is-dismissable` class, I simply added `style="padding-right:38px; position:relative;"` which does the same thing, and then I added the following markup to produce the button itself.

```html
<div class="notice" style="padding-right:38px; position:relative;">
    <p>Notice content here.</p>
    <a class="notice-dismiss" href="[URL to dismiss]"><span class="screen-reader-text">Dismiss</span></a>
</div>
```

**Comment by kristineds** [comment]
@jaswsinc How do I replace the content on the `<p>` tag with the StCR notice? https://github.com/websharks/comment-mail-pro/commit/d98f1924a4ca2f0f4cd9de65e8a47c31d6d897f2

I'm looking at `$notice` on [this file](https://github.com/websharks/comment-mail-pro/blob/433c3acb15bd3d35ec710c6ce13535f2a7dedb86/src/includes/functions/stcr.php#L82): 

screen shot 2016-02-26 at 1 45 07 am

**Comment by raamdev** [comment]
@kristineds writes...
> How do I replace the content on the <p> tag with the StCR notice?

I don't understand the question. Could you clarify for me? :-)

**Comment by kristineds** [comment]
@raamdev With the PR that I submitted, I copied the HTML source for the"Plugin activated" notice as you pointed out here: https://github.com/websharks/comment-mail/issues/193#issue-123934285

However, I'm not sure how to integrate it with the StCR notice above :arrow_up: . Could you point me in the right direction?  Here is what I have so far.. https://github.com/websharks/comment-mail-pro/commit/d98f1924a4ca2f0f4cd9de65e8a47c31d6d897f2

**Comment by raamdev** [comment]
@kristineds The `enqueueNotice()` method queues up the notices and stores them in the database so that they can be displayed. A single `allAdminNotices()` method in `src/includes/classes/Plugin.php` is used to display the notices. That's where the dismiss button style and markup is generated. You'll need to adjust that method so that it uses the newer dismiss button styling.

_Note: It took me a little while to wrap my head around how all of these notice utilities work the first time I looked at this code, but once you follow the routines and figure out how they call each other, it all makes sense (and it's all very flexible!). I suggest spending some time tracing the various calls in the source code to understand how they eventually get displayed, e.g., start with `enqueueNotice()` and then figure out what that calls, and so on._

**Comment by kristineds** [comment]
@raamdev @jaswsinc Submitting my PR for this. https://github.com/websharks/comment-mail-pro/pull/59/commits/8e92ff9805aa40ab7c42581dea14e52de7346bcb

screen shot 2016-03-16 at 11 46 10 pm

File: src/includes/classes/Plugin.php [hunk 1]
```diff
@@ -2280,7 +2276,7 @@ public function allAdminNotices()
                         $_classes .= ' updated'; // Green informational notice
                 }
                 $_full_markup = // Put together the full markup; including other pieces.
-                    '<div class="'.esc_attr($_classes).' style="clear: both;">'.// clear:both needed to fix StCR options page clash; see http://bit.ly/1V83vQl
+                    '<div class="notice '.esc_attr($_classes).'" style="clear: both; padding-right: 38px; position: relative;">'.// clear:both needed to fix StCR options page clash; see http://bit.ly/1V83vQl
```
Comment (raamdev) [inline src/includes/classes/Plugin.php]: @kristineds Please move the extra `style=""` code on this line back into `$_dismiss_style` and then use that variable here instead. That way, if the notice being displayed is _not_ dismissable (i.e., doesn't have a dismiss button), we don't use that extra CSS that is used when that button is present (you'll need to add `$_dismiss_style = '';` as the default value, same as `$_dismiss`).

---

Also, I tested this with a short notice and I see that there isn't enough spacing on the bottom:



I suggest adding `padding-bottom: 10px;` to this div (that _can_ be hardcoded, as it's applicable even when there's no dismiss button).

**Comment by kristineds** [comment]
@raamdev Like this? https://github.com/websharks/comment-mail-pro/pull/59/commits/6c43f439550c4d186522ec9cc76a4265db5b0eb5

screen shot 2016-03-17 at 12 58 59 am

**Comment by raamdev** [comment]
@kristineds Yes, that looks great! All you're missing now is setting the `$_dismiss_style` to a blank default, so that those styles don't get applied when not using a dismiss button. It should go right under this line: https://github.com/websharks/comment-mail-pro/blob/feature/193/src/includes/classes/Plugin.php#L2264

**Comment by kristineds** [comment]
@raamdev Thanks! I added it here. https://github.com/websharks/comment-mail-pro/pull/59/commits/2b87b27f2c137b6dbe5ab5fd00cac3a315644d6e

**Comment by raamdev** [comment]
@kristineds Thank you. I just tested this again and it looks like I was wrong about needing `padding-bottom: 10px;` for all of the notice types:



So, it looks like we only need that extra padding on the bottom when we're showing a dismiss button, so you can move that CSS into the `$_dismiss_style` variable.

Once you've changed that, please post a screenshot showing the Dashboard notices as I did above so we can confirm the padding looks right on both types.

**Comment by kristineds** [comment]
@raamdev Like this? https://github.com/websharks/comment-mail-pro/pull/59/commits/16ddeff5dcb2998c78e924aa5f10f0d0469dc452

screen shot 2016-03-18 at 6 22 18 pm

**Comment by raamdev** [comment]
@kristineds Yes, I just tested this and it looks great now. Thank you! :smile:

**Pull request closed without merging**